data.py

```diff
@@ -58,13 +59,153 @@ def jitter_pointcloud(pointcloud, sigma=0.01, clip=0.05):
+
+        masked_pointcloud = pointcloud[boolean_mask, :]
+
+        if len(masked_pointcloud) == 0:
+            return pointcloud
+
+        while masked_pointcloud.shape[0] < original_length:
+            masked_pointcloud = np.concatenate([masked_pointcloud, masked_pointcloud], axis=0)
+
+        masked_pointcloud = masked_pointcloud[:original_length, :]
+        return masked_pointcloud
+
+
+class RandomRemoveEightQuadrant:
+    """
+    Remove an eighth of the pointcloud that belongs to a quadrant
+    """
+    def __call__(self, pointcloud):
+        original_length = pointcloud.shape[0]
+
+        larger_x = np.random.randint(2)
+        larger_y = np.random.randint(2)
+        larger_z = np.random.randint(2)
+
+        boolean_arr_x = pointcloud[:, 0] > 0 if larger_x else pointcloud[:, 0] < 0
+        boolean_arr_y = pointcloud[:, 1] > 0 if larger_y else pointcloud[:, 1] < 0
+        boolean_arr_z = pointcloud[:, 2] > 0 if larger_z else pointcloud[:, 2] < 0
+
+        masked_out = np.logical_and(boolean_arr_x, boolean_arr_y)
+        masked_out = np.logical_and(masked_out, boolean_arr_z)
+
+        masked_pointcloud = pointcloud[np.logical_not(masked_out), :]
+
+        if len(masked_pointcloud) == 0:
+            return pointcloud
+
+        while masked_pointcloud.shape[0] < original_length:
+            masked_pointcloud = np.concatenate([masked_pointcloud, masked_pointcloud], axis=0)
+
+        masked_pointcloud = masked_pointcloud[:original_length, :]
+
+        return masked_pointcloud
+
+
+class Augment:
+    def __init__(
+            self,
+            transforms: Tuple[Any] = (
+                Jitter(),  # TODO: ALWAYS LEAVE THIS LINE UNCOMMENTED
+                RandomRemoveTwoQuadrant(),  # TODO: Sergi: Uncomment this line and comment out the rest, run python command
+                # RandomRemoveFourQuadrant(),  # TODO: Eric: Uncomment this line, and comment out the rest and run the python command.
+                # RandomRemoveEightQuadrant(),  # TODO: Siddhant: Uncomment this line and comment out the rest and run the python command.
+                # RandomRemove()  # TODO: Arvind: Uncomment this line and comment out the rest and run the python command.
+            )
+         ):
+        self._transforms = transforms
+    def __call__(self, pointcloud):
+        for transform in self._transforms:
+            pointcloud = transform(pointcloud)
+
+        return pointcloud
+
+
 class ModelNet40(Dataset):
-    def __init__(self, num_points, partition='train', gaussian_noise=False, unseen=False, factor=4):
+    def __init__(self,
+                 num_points,
+                 partition='train',
+                 gaussian_noise=False,
+                 unseen=False,
+                 augment: Optional[Augment] = Augment(),
+                 is_augment_a: bool = True,  # We are only augmenting the source point cloud throughout the experiments.
```

Comment:
Throughout the experiments, we are only augmenting the first point cloud in order to simulate real life/empirical occlusions in our dataset. Make sure to keep the constructor arguments as is